source/ShopPanel.cpp

```diff
@@ -1220,7 +1222,16 @@ void ShopPanel::SideSelect(Ship *ship)
 		}
 	}
 	else if(!control)
+	{
 		playerShips.clear();
+		if(clicks > 1)
+			for(const shared_ptr<Ship> &it : player.Ships())
+			{
+				Ship *itShip = it.get();
+				if(itShip != ship && itShip->Immitates(*ship))
+					playerShips.insert(itShip);
+			}
+	}
```

Comment:
Can we support holding control to preserve the existing selection and adding the new ships to it?
Maybe move the "if(clicks > 1)` block below this if-else-if block so we have:
```c++
else if(!control)
	playerShips.clear();
else if(clicks == 1 && playerShips.contains(ship))
{
	// Whatever is currently in there.
}
if(clicks > 1)
{
	// New stuff.
}
```

Reply:
yeah sure but what about shift?

Reply:
Not sure about shift. I don't think it has an immediately obvious meaning in this context, whereas control generally means "add this to the existing selection" (except if this item is already selected). If double clicking means "select every ship like this", I think it is intuitive to have control double clicking mean "add every ship like this to the selection".

Interestingly, shift clicking the most recently shift clicked ship selects everything from that ship up to the end of the list, which is probably not desired.

Reply:
OK I'll exclude shift. But ctrl can also deselect already selected ships, so maybe we should check if all similar ships are selected and then deselect them, otherwise add to the selection?

Reply:
Yeah, that sounds good.